File: Algorithms/Implementation/BonAppetit.java
package Implementation;

import java.util.Scanner;

/**
 *	Algorithms -> Implementations -> Bon Appétit
 *	Easy
 */
public class BonAppetit {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
        int numberOfITems = in.nextInt();
        int indecOFItemNotEaten = in.nextInt();
        int[] foodCostsArry = new int[numberOfITems];
        int realFoodCosts = 0;
        for(int i = 0; i < numberOfITems; i ++) {
            foodCostsArry[i] = in.nextInt();
            if(i != indecOFItemNotEaten) realFoodCosts += foodCostsArry[i];
        }
        int amountCharged = in.nextInt();
        int amountToCharge = realFoodCosts/2;
        if(amountCharged == amountToCharge) {
            System.out.println("Bon Appetit");   
        } else {
            System.out.println(amountCharged - amountToCharge);
        }
        in.close();
	}

}

File: Algorithms/Implementation/BonAppetit.c
#include <stdio.h>

/**
 *	Algorithms -> Implementation -> Bon Appétit
 *	Easy
 */

int main() {
	int numberOfItems, indexOfItemNotEaten;

	scanf("%d %d", &numberOfItems, &indexOfItemNotEaten);

	int foodCosts[numberOfItems];

	int realFoodCosts = 0;

	for(int i = 0; i < numberOfItems; i++) {
		scanf("%d", &foodCosts[i]);
		if(i != indexOfItemNotEaten) realFoodCosts += foodCosts[i];
	}

	int amountCharged;

	scanf("%d", &amountCharged);

	int amountToCharge = realFoodCosts/2;

	if(amountCharged == amountToCharge) {
		printf("Bon Appetit");
	} else {
		printf("%d", amountCharged - amountToCharge);
	}

	return 0;
}
